File: ios/Khrysalis Template/com/lightningkite/khrysalistemplate/api/APIOnline.shared.swift
//Package: com.lightningkite.khrysalistemplate.api
//Converted using Khrysalis2

import Foundation
import Khrysalis
import RxSwift
import RxRelay
import KhrysalisMaps



public class APIOnline: APIInterface {
    
    public var baseUrl: String
    
    
    public func getExamplePosts(onResult: @escaping (Int32, Array<Post>?, String?) -> Void) -> Void {
        HttpClient.call(url: baseUrl + "/posts", method: HttpClient.GET, headers: [:], body: nil, onResult: onResult)
    }
    
    public init(baseUrl: String = "https://jsonplaceholder.typicode.com") {
        self.baseUrl = baseUrl
    }
    convenience public init(_ baseUrl: String) {
        self.init(baseUrl: baseUrl)
    }
}
 

File: android/src/main/java/com/lightningkite/khrysalistemplate/api/APIOnline.shared.kt
///Kotlin Only
@file:Suppress("ConvertToStringTemplate", "RemoveExplicitTypeArguments")

///End Kotlin Only

package com.lightningkite.khrysalistemplate.api

import com.lightningkite.khrysalis.net.HttpClient
import com.lightningkite.khrysalis.escaping
import com.lightningkite.khrysalis.net.readJson
import com.lightningkite.khrysalistemplate.models.Post
import io.reactivex.Single


class APIOnline(val baseUrl: String = "https://jsonplaceholder.typicode.com") : APIInterface {
    override fun getExamplePosts(): Single<List<Post>> {
        return HttpClient.call(
            url = baseUrl + "/posts",
            method = HttpClient.GET,
            headers = mapOf()
        ).readJson()
    }

}
